components/dashboard/challenge-card.tsx

```
@@ -19,38 +20,141 @@ export function ChallengeCard({
+				</div>
+
+				{/* Action button - responsive sizing */}
+				<div className="flex-shrink-0 w-full sm:w-auto">
+					<button
+						className={cn(
+							"w-full sm:w-auto py-2 px-4 text-xs sm:text-sm font-medium rounded-md transition-all duration-200 active:scale-95",
+							status === "active"
+								? "bg-blue-600 hover:bg-blue-700 text-white shadow-sm hover:shadow-md"
+								: "bg-gray-100 dark:bg-gray-800 hover:bg-gray-200 dark:hover:bg-gray-700 text-gray-900 dark:text-white"
+						)}
+					>
```

Biome (1.9.4)

[error] Provide an explicit type prop for the button element.

The default type of a button is submit, which causes the submission of a form when placed inside a `form` element. This is likely not the behaviour that you want inside a React application.
Allowed button types are: submit, button or reset

(lint/a11y/useButtonType)